### ISSUE TITLE
Bad detection of a Smarty file

### ISSUE THREAD

**Issue by psa-jforestier** [AlDanial/cloc#17]
I have a directory full of ".smarty" files and PHP files.
Calling cloc on this directory do not report smarty file, only php files. Only lines of code , comments ect of the PHP files are reported. Smarty files are ignored.

To reproduce : 
- create a .PHP file with only PHP marker inside
- create a dummy SMARTY file containing somethig like :
```
{if $cta.style eq 'STYLE_DROPDOWNLIST'}
    <div class="list{if isset($cta.dimension)} grid-col span_{$cta.dimension}_of_12{/if}">
    </div>
{elseif $cta.style eq 'STYLE_SIMPLELINK'}
   <h1>HELLO WORLD</h1>
{else}
      <h1>HELLO WORLD 2</h1>
{/if}
```
- start cloc
It says 
       2 text files.
       2 unique files.
       1 file ignored.
And report only PHP stats.

**Comment by AlDanial** [comment]
Odd; the code detects Smarty files internally, then ignores them.

**Comment by AlDanial** [comment]
Fixed with git commit 1604cfa
